### PR TITLE
Cortex-M52 experimental

### PR DESCRIPTION
**Pull request by jkrech**
Add experimental support for Cortex-M52 in AC6

### CODE REVIEW

File: tools/buildmgr/cbuildgen/config/AC6.6.18.0.cmake [hunk 1]
```diff
@@ -370,6 +373,32 @@ elseif(CPU STREQUAL "Cortex-M35P")
...
+    endif()
+  else()
+    if(FPU STREQUAL "DP_FPU")
+      set(ARMCLANG_CPU "-mcpu=Cortex-M52 -mfloat-abi=hard")  
+    elseif(FPU STREQUAL "SP_FPU")
+      message(FATAL_ERROR "Error: Cortex-M52+nofp.dp is not supported!")
+	  else()
+      message(FATAL_ERROR "Error: Cortex-M52+nofp is not supported!")
+	  endif()
+  endif()
```
[the remark this answers is not in the document]
Comment (jkrech) [reply tools/buildmgr/cbuildgen/config/AC6.6.18.0.cmake]: I have replaced tabs with spaces to correct the indentation. Sorry.

**Comment by jkrech** [comment]
I have added a space for ARMASM_CPU for Cortex-M52 so the variable is defined but not specifying `--cpu=...`

File: tools/buildmgr/cbuildgen/config/AC6.6.18.0.cmake [hunk 2]
```diff
@@ -370,6 +373,32 @@ elseif(CPU STREQUAL "Cortex-M35P")
...
+    elseif(FPU STREQUAL "SP_FPU")
+      set(ARMCLANG_CPU "-mcpu=Cortex-M52+nomve+nofp.dp -mfloat-abi=hard")
+    else()
+      message(FATAL_ERROR "Error: Cortex-M52+nomve+nofp is not supported!")
+    endif()
+  elseif(MVE STREQUAL "INT_MVE")
+    if(FPU STREQUAL "DP_FPU")
+      set(ARMCLANG_CPU "-mcpu=Cortex-M52+nomve.fp -mfloat-abi=hard")
+    elseif(FPU STREQUAL "SP_FPU")
+      message(FATAL_ERROR "Error: Cortex-M52+nomve+nofp.dp is not supported!")
```
Comment (brondani) [inline tools/buildmgr/cbuildgen/config/AC6.6.18.0.cmake:389]:
```suggestion
    if(FPU STREQUAL "SP_FPU")
      set(ARMCLANG_CPU "-mcpu=Cortex-M52+nomve.fp+nofp.dp -mfloat-abi=hard")
    elseif(FPU STREQUAL "DP_FPU")
      message(FATAL_ERROR "Error: Cortex-M52+nomve.fp is not supported!")
```

**Review by brondani — APPROVED** [review]
LGTM

**Pull request merged**